For clarification, if the local base64 version matches hawk's one, it compiles, if they don't match anymore one of them get updated, then you'll get 
```
error[E0277]: the trait bound `hawk::Error: From<DecodeError>` is not satisfied
```

--- a/src/bewit.rs
+++ b/src/bewit.rs
@@ -81,7 +81,7 @@ const BACKSLASH: u8 = b'\\';
 impl<'a> FromStr for Bewit<'a> {
     type Err = Error;
     fn from_str(bewit: &str) -> Result<Bewit<'a>> {
-        let bewit = base64::decode(bewit).map_err(|e| Error::from_base64_error(e))?;
+        let bewit = base64::decode(bewit).map_err(Error::from_base64_error)?;
 
         let parts: Vec<&[u8]> = bewit.split(|c| *c == BACKSLASH).collect();
         if parts.len() != 4 {

--- a/src/bewit.rs
+++ b/src/bewit.rs
@@ -81,7 +81,7 @@ const BACKSLASH: u8 = b'\\';
 impl<'a> FromStr for Bewit<'a> {
     type Err = Error;
     fn from_str(bewit: &str) -> Result<Bewit<'a>> {
-        let bewit = base64::decode(bewit)?;
+        let bewit = base64::decode(bewit).map_err(|e| Error::from_base64_error(e))?;
 
         let parts: Vec<&[u8]> = bewit.split(|c| *c == BACKSLASH).collect();
         if parts.len() != 4 {

--- a/src/error.rs
+++ b/src/error.rs
@@ -45,8 +45,10 @@ pub enum InvalidBewit {
     Ext,
 }
 
-impl From<base64::DecodeError> for Error {
-    fn from(e: base64::DecodeError) -> Self {
+impl Error {
+    // this cannot be a `From<..>` implementation as that publicly exposes the version of base64
+    // used in this crate.
+    pub(crate) fn from_base64_error(e: base64::DecodeError) -> Self {
         Error::Decode(e.to_string())
     }
 }

